xml/array.go

```diff
@@ -0,0 +1,66 @@
+package xml
+
+import (
+	"bytes"
+)
+
+// arrayKey is the default member wrapper tag name for XML Array type
+const arrayKey = "member"
+
+// Array represents the encoding of a XML array type
+type Array struct {
+	w          *bytes.Buffer
+	scratch    *[]byte
+	openTagFn  func()
+	closeTagFn func()
+}
+
+// newArray returns an array encoder which sets the default array
+// member wrapper to `member`.
+//
+// for eg. an array ["value1", "value2"] is represented as
+// <List><member>value1</member><member>value2</member></List>
+func newArray(w *bytes.Buffer, scratch *[]byte) *Array {
+	var openTagFn = func() {
```

Comment:
can `newArray` and `newArrayWithCustomName` constructors be squashed into one? They both look the same with the only difference being the tag name. Since the callers of these two already are separated into regular array and custom array that level could pass in the tag name as needed.